src/server/zset_family.cc

```diff
@@ -130,6 +133,46 @@ OpResult<PrimeIterator> FindZEntry(const ZParams& zparams, const OpArgs& op_args
   return it;
 }
 
+// buf must be at least 12 bytes long.
+bool ToAsciiGeoHash(const std::optional<double>& val, char buf[]) {
+  if (!val)
+    return false;
+
+  double score = *val;
+
+  double xy[2];
```

Comment:
Same here and for `r` - I'd use `std::array<>` which is safer

Reply:
I do not think here it matters. i pass `xy` to a c function, and i call hardcoded indices elsewhere. why do you think array is safer?

Reply:
`std::array`s are generally safer, and since they have no real downsides (expect for longer names?), I think they should be used as the default over c-style arrays.
Specifically, I can foresee a refactor that will make the compiler unaware of the size of the array (like when passed to a function), and `std::array` will solve that.
Feel free to disagree, as always.